crypto/src/hash/traits.rs

```diff
@@ -36,6 +36,12 @@ where
     }
 }
 
+/// Trait that converts given generic value into either field(s) or bits(s)
+/// and append to mutable [mina_signer::ROInput], this trait should be implemented for [mina_signer::ROInput]
+pub trait RandomOraclePartialInput<T> {
```

Comment:
It is not clear what this trait is meant to provide that ROInput doesn't already.  Can you please explain?

With respect to the concept of "partial" random oracle input, there should be no partial implementation of the interface-- anything hashed of signed requires both `to_roinput` and `domain_string`.  If we find something in the ocaml that doesn't have both, then it is probably a bug.

Again you should hold off on this implementation until we update our Signable and Hashable interfaces for you.

Reply:
In the ocaml code, every field type of the `Account` data structure has `to_input` function, and the `to_input` of `Account` simply merges `to_input` result of every field, this it flexible to allow an arbitrary combination of the field types into a new data structure (if needed) whose `to_input` method can be easily implemented. 

An alternative solution could be, implementing `Hashable` for each field type, and then ues another API to merge multiple `ROInput`s into one. Does that make sense?

Reply:
Done.